### DISCUSSION TITLE
Ability to automatically avoid naming collisions

### DISCUSSION THREAD

**Discussion by benjam-es** [spatie/laravel-medialibrary#3189] [Ideas]
So prevent masses of directories, I use a custom path generator. In my case it is ../year/month/name.jpg

e.g. ``../2023/02/spatie-logo.png``

As users often upload the same image/pdf to multiple models, this can lead to the first being overwritten. 

Could a config be added to 'check if file exists' and throw an error, or possibly a config to auto increment file name if a collision occurs. i.e. if ``../2023/02/spatie-logo.png`` exists, it will try ``../2023/02/spatie-logo-1.png`` then ``../2023/02/spatie-logo-2.png`` etc

**Comment by parallels999** [comment]
If you use custom path generator avoid naming collisions is at your own risk
This package use uuid for avoid naming collisions natively, so, use uuid too

e.g. `../2023/02/123e4567-e89b-12d3-a456-426614174000.png`
